File: advent-of-code-2023/day2.cpp
# include <iostream>
# include <vector>

using namespace std;

int main(){
    string game;

    long long id_c = 0;
    long long power = 0;
    int id = 1;
    while(getline(cin, game)){
        if(game.empty()) {
            break;
        }

        long long red=0, green=0, blue=0;
        int i = 0;

        while(game[i++] != ':');
        i++;
        for(; i<game.length(); i++){
            
            
            if(game[i] >= '0' && game[i] <= '9' && game[i+1] != ':'){
                long long num = 0;
                while(game[i] != ' '){
                    num = num * 10ll + (game[i]-'0');
                    i++;
                }


                if(game[i+1] == 'r'){
                    cout << "enterd into red";
                    red = max(red, num);
                } else if(game[i+1] == 'g'){
                    green = max(green, num);
                } else if(game[i+1] == 'b'){
                    blue = max(blue, num);
                }
            }
        }

    cout << "red = " << red << " green " << green << " blue = " << blue << endl; 
        if(red <= 12 && green <= 13 && blue <= 14){
            id_c += id;
        }
        id++;
        power += red * green * blue;
    }

    cout <<  "total id_c = " << id_c << endl;
    cout << "power = " << power << endl;
}

File: advent-of-code-2023/aoc_py/day2.py
with open('input') as f:
    lines = f.readlines()

total = 0
total_2 = 0
test = {'red': 12, 'green': 13, 'blue':14}
for line in lines:
    game, plays = line.strip().split(':')
    actual = {'red': 0, 'green': 0, 'blue':0}
    for play in plays.split(';'):
        for color in play.split(','):
            color_data = color.strip().split()
            actual[color_data[1]] = max(actual[color_data[1]], int(color_data[0]))
    
    flag = 1
    power_cube = 1
    for color_key in test.keys():
        power_cube *= actual[color_key]
        if test[color_key] >= actual[color_key]:
            flag &= 1
        else:
            flag &= 0

    if flag:
        total += int(game.split()[1])

    # part 2
    total_2 += power_cube

print("part 1: ", total)
print("part 2: ", total_2)
